### ISSUE TITLE
Wrong indexOf type in array

### ISSUE THREAD

**Issue by aczekajski** [Microsoft/TypeScript#31022]
**TypeScript Version:**  3.4.0 and 3.5.0-dev.20190418

**Search Terms:** indexOf, array union, ts(2345)

**Code**
There seems to be an error in type of `indexOf` function in `Array` interface. It seems to be too strict, not allowing to search for values that might or might not be inside the array.
```ts
type AB = 'A' | 'B';
type CD = 'C' | 'D';
type ABCD = AB | CD;

const fn1 = (t: AB[], v: ABCD) => t.indexOf(v); // v might be "B" so it is a perfectly valid indexOf usage

const fn2 = (t: AB[], v: CD) => t.indexOf(v); // AB and CD have nothing in common, this might be an error

const fn3 = (t: ABCD[], v: AB) => t.indexOf(v); // this is ok
```

**Expected behavior:**
`indexOf` inside `fn1` shouldn't throw an error because there is a non-empty intersection between array values type and searched value type

**Actual behavior:**
`indexOf` inside `fn1` throws the following error:
```
Argument of type 'ABCD' is not assignable to parameter of type 'AB'.
  Type '"C"' is not assignable to type 'AB'.ts(2345)
```

**Playground Link:** [TypeScript Playground](https://www.typescriptlang.org/play/#src=type%20AB%20%3D%20'A'%20%7C%20'B'%3B%0D%0Atype%20CD%20%3D%20'C'%20%7C%20'D'%3B%0D%0Atype%20ABCD%20%3D%20AB%20%7C%20CD%3B%0D%0A%0D%0Aconst%20fn1%20%3D%20(t%3A%20AB%5B%5D%2C%20v%3A%20ABCD)%20%3D%3E%20t.indexOf(v)%3B%20%2F%2F%20v%20might%20be%20%22B%22%20so%20it%20is%20a%20perfectly%20valid%20indexOf%20usage%0D%0A%0D%0Aconst%20fn2%20%3D%20(t%3A%20AB%5B%5D%2C%20v%3A%20CD)%20%3D%3E%20t.indexOf(v)%3B%20%2F%2F%20AB%20and%20CD%20have%20nothing%20in%20common%2C%20this%20might%20be%20an%20error%0D%0A%0D%0Aconst%20fn3%20%3D%20(t%3A%20ABCD%5B%5D%2C%20v%3A%20AB)%20%3D%3E%20t.indexOf(v)%3B%20%2F%2F%20this%20is%20ok)

**Comment by RyanCavanaugh** [comment]
Effective duplicate of #26255
